jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/common/Icon.java

```diff
@@ -2,171 +2,67 @@
-                new BufferedReader(
-                        new InputStreamReader(
-                                new URL("https://raw.githubusercontent.com/Templarian/MaterialDesign-JS/master/mdi.js")
-                                        .openConnection()
-                                        .getInputStream())
-                ).lines().forEach(line -> {
-                    Matcher mdiNameMatch = mdiName.matcher(line);
-                    Matcher mdiPathMatch = mdiPath.matcher(line);
-                    if (mdiNameMatch.find() && mdiPathMatch.find()) {
-                        String name = mdiNameMatch.group().substring(3).replaceAll("(\\B[A-Z])|(\\B[0-9]+)", "_$1$2").toUpperCase();
-                        map.put(name, mdiPathMatch.group(1));
-                    }
-                });
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }
-        return map;
-    }
-
-    @JDIAction("Get '{name}' mdi name")
-    public String getMdiIconName() {
-        Pattern iconPtn = Pattern.compile("(mdi-[\\w-]*)|(fa-[\\w-]*)|material-icons.*>(.*)<|path d=\"(.*)\"");
-        Matcher matcher = iconPtn.matcher(core().getAttribute("outerHTML"));
-
-        String name = null;
-        if (matcher.find()) {
-            if (matcher.group(1) != null) {
-                name = matcher.group(1).substring(4).replace("-", "_").toUpperCase();
-            } else if (matcher.group(3) != null) {
-                name = matcher.group(3).toUpperCase();
-            } else if (matcher.group(4) != null) {
-                name = getMdiMap().getKey(matcher.group(4));
-            } else {
-                throw new IllegalStateException(attr("class") + " is not Material Design Icon");
-            }
-        }
-        return name;
-    }
+public class Icon extends UIBaseElement<IconAssert> implements HasClick, HasLabel {
 
     @JDIAction("Get '{name}' type")
     private String getType() {
         String res = null;
         String iconClass = core().getAttribute("class");
         if (iconClass.contains("mdi-")) {
             res = Arrays.stream(iconClass.split(" "))
-                    .filter(s -> s.startsWith("mdi-")).collect(Collectors.joining());
+                .filter(s -> s.startsWith("mdi-")).collect(Collectors.joining());
         } else if (iconClass.contains("fa-")) {
             res = Arrays.stream(iconClass.split(" "))
-                    .filter(s -> s.startsWith("fa-")).collect(Collectors.joining());
+                .filter(s -> s.startsWith("fa-")).collect(Collectors.joining());
         } else if (iconClass.contains("material-icons")) {
             res = core().getText();
-        } else if (core().getAttribute("innerHTML").contains("svg")) {
+        } else if (iconClass.contains("v-icon notranslate")) {
             res = "svg icon";
         }
         return res;
     }
 
-    @JDIAction("Get '{name}' color")
-    public String color() {
+    @JDIAction("'{name}' has color")
```

Comment:
Looks like it is a part of interface HasColor

Reply:
Resolved